### DISCUSSION TITLE
Is there a way to pass in a unique ID to the options of the Select component?

### DISCUSSION THREAD

**Discussion by jasnlink** [Shopify/polaris-react#11111] [General] [answered]
There are cases where multiple options of different labelling use the same underlying value such as in the case of country calling codes:

- **Label:** Canada (+1), **Value:** +1
- **Label:** United States (+1), **Value:** +1

Because the underlying component simply grabs the value and uses it as a key for React, it is causing a lot of duplicate key errors:

<img width="1090" alt="image" src="https://github.com/Shopify/polaris/assets/88203684/289893ca-c5d4-494a-bf0b-e37ebf8f35c5">

I could potentially solve this problem by inserting the whole label as a value and then extracting the calling code value on form submit/validation, but that just seems overkill for a simple Select component.

**Comment by yurm04** [comment, marked as answer]
Hey @jasnlink thanks for reaching out. I've merged in a fix to support this use case. You can now pass in an optional `key` property to your option object that will override the default.

It will be included in the next minor release of Polaris 🙏 https://github.com/Shopify/polaris/pull/11172

**Reply by jasnlink** [reply]
Love to see this, thanks 🙏
